doc/developer/scan.txt

```diff
@@ -132,22 +63,266 @@ files, and section of the code they deal with, are :
+outer function graph. The inputs and outputs of scan's inner function are
+designated "inner inputs" and "inner outputs".
+
+
+Scan variables
+==============
+
+The following are the different types of variables that Scan has the capacity to
+handle, along with their various caracteristics.
+
+**Sequence** : A sequence is a Theano variable which Scan will iterate over and
+give sub-elements to its inner function as input. A sequence has no associated
+output. For a sequence variable `X`, at timestep `t`, the inner function will
+receive as input the sequence element `X[t]`. These variables are used through
+the argument `sequences` of the `theano.scan()` function.
+
+**Non-sequences** : A non-sequence is a Theano variable which Scan will provide
+`as-is` to its inner function. Like a sequence, a non-sequence has no
+associated output. For a non-sequence variable X, at timestep `t`, the inner
+function will receive as input the variable X. These variables are used through
+the argument `non_sequences` of the `theano.scan()` function.
+
+**Nitsot (no input tap, single output tap)** : A nitsot is an output variable of
+the inner function that is not fed back as an input to the next iteration of the
+inner function. Nitsots are typically encountered in situations where Scan is
+used to perform a 'map' operation (every element in a tensor is independently
+altered using a given operation to produce a new tensor) such as squaring every
+number in a vector.
+
+**Sitsot (single input tap, single output tap)** : A sitsot is an output
+variable of the inner function that is fed back as an input to the next
+iteration of the inner function. A typical setting where a sitsot might be
+encountered is the case where scan is used to compute the cumulative sum over
+the elements of a vector and a sitsot output is employed to act as an
+accumulator.
+
+**Mitsot (multiple input taps, single output tap)** : A mitsot is an output
+variable of the inner function that is fed back as an input to future iterations
+of the inner function (either multiple future iterations or a single one that
+isn't the immediate next one). For example, a mitsot might be used in the case
+where scan is used to compute the Fibonacci sequence, one term of the sequence
+at every timestep, since every computed term needs to be reused to compute
+the two next terms of the sequence.
+
+**Mitmot (multiple input taps, multiple output taps)** : These outputs exist
+but they cannot be directly created by the user. They can appear in a theano
+graph as a result of taking the gradient of the output of a Scan : taking the
+gradient of the output of a Scan wrt its inputs will result in the creation of
+a new Scan node used to compute the gradients of the first Scan node. If the
+original Scan had sitsots or mitsots variables, the new Scan will use mitmots
+to compute the gradients through time for these variables.
+
+
+To synthesise :
```

Comment:
synthesize

I think the spelling with an `s` is the British (maybe also Canadian) spelling. I don't know if we want the doc to be "standard"/U.S. english?